ddsp/training/docker/user_interaction.py

```diff
@@ -0,0 +1,150 @@
+  if steps_per_save == '':
+    steps_per_save = '300'
+  input_dict['steps_per_save'] = steps_per_save
+
+  steps_per_summary = input('\nInsert the number of steps per summary '
+                            'or skip for the default value (300): ')
+  if steps_per_summary == '':
+    steps_per_summary = '300'
+  input_dict['steps_per_summary'] = steps_per_summary
+
+  early_stop_loss_value = input('\nInsert the early stop loss value '
+                                'or skip for the default value (5): ')
+  if early_stop_loss_value == '':
+    early_stop_loss_value = '5'
+  input_dict['early_stop_loss_value'] = early_stop_loss_value
+
+  return input_dict
+
+def build_image(input_dict):
+  """Builds the docker image"""
+  build_command = 'docker build -f Dockerfile -t '\
+                  + input_dict['image_uri'] + ' ./'
+  os.system(build_command)
+
+def push_image(input_dict):
+  """Pushes the docker image on Google Cloud Registry"""
+  pushing_image = 'docker push ' + input_dict['image_uri']
+  os.system(pushing_image)
+
+def submit_job(input_dict):
+  """Submits the job on AI Platform"""
+  os.system('export PATH=/usr/local/google/home/$USER/.local/bin:$PATH')
+
+  submitting_job = 'gcloud beta ai-platform jobs submit training '\
+  + input_dict['job_name'] + ' --region ' + input_dict['region']\
+  + ' --master-image-uri ' + input_dict['image_uri']\
+  + ' --config ' + input_dict['config_path']\
+  + ' -- --save_dir=' + input_dict['save_dir']\
+  + ' --restore_dir=' + input_dict['restore_dir']\
+  + ' --file_pattern=' + input_dict['data_path'] + '/train.tfrecord*'\
+  + ' --batch_size=' + input_dict['batch_size']\
+  + ' --learning_rate=' + input_dict['learning_rate']\
+  + ' --num_steps=' + input_dict['no_of_steps']\
+  + ' --steps_per_summary=' + input_dict['steps_per_summary']\
+  + ' --steps_per_save=' + input_dict['steps_per_save']\
+  + ' --early_stop_loss_value=' + input_dict['early_stop_loss_value']
+  os.system(submitting_job)
+
+def enable_tensorboard(input_dict):
+  """Enables Tensorboard"""
+  os.system('gcloud auth login')
+  tensorboard_command = 'tensorboard --logdir='\
+                        + input_dict['save_dir'] + ' --port=8082 &'
+  os.system(tensorboard_command)
+
+def upload_logs(input_dict):
+  """Uploads logs to TensorBoard.dev"""
+  tensorboard_dev_command = 'tensorboard dev upload --logdir '\
+                            + input_dict['save_dir']\
+  + ' --name \"' + input_dict['job_name'] + '\"'
+  os.system(tensorboard_dev_command)
+
+def main():
+  """Gathers input, submits job and enables TensorBoard"""
+  input_dict = get_input()
+
+  build_image(input_dict)
+  print('Docker image built')
+
+  push_image(input_dict)
+  print('Image pushed to Google Cloud Registry')
+
+  submit_job(input_dict)
+  print('Job submitted to AI Platform')
+
+  enable_tensorboard(input_dict)
+  print('Tensorboard enabled')
+
+  upload_logs(input_dict)
+  print('Logs uploaded to TensorBoard.dev')
+
+if __name__ == '__main__':
```

Comment:
As I've said above, I'd follow ddsp_run in the approach here, using console_entry_point() run() and absl for app and flag management

Reply:
Done